app/controllers/compute_resources_vms_controller.rb

```diff
@@ -1,11 +1,17 @@
 class ComputeResourcesVmsController < ApplicationController
   def index
     @compute_resource = find_compute_resource(:view_compute_resources_vms)
-    @vms = @compute_resource.vms.all(params[:filters] || {})
+    load_vms
     @authorizer = Authorizer.new(User.current, :collection => [@compute_resource])
     respond_to do |format|
       format.html
-      format.json { render :json => @vms }
+      format.json do
+        if @compute_resource.supports_vms_pagination?
+          render :partial => "compute_resources_vms/index/#{@compute_resource.provider.downcase}.json"
+        else
+          render :json => @vms
```

Comment:
I think I will remove the default json as this will break this path anyways for ovirt in case anyone uses it (I could not find any such case anyways).